src/shell/shell_misc.cpp

```diff
@@ -467,16 +462,16 @@ bool DOS_Shell::Execute(std::string_view name, std::string_view args)
 		/* delete old batch file if call is not active*/
 		/*keep the current echostate (as delete bf might change it )*/
 		const bool temp_echo = echo;
-		if (bf && !call) {
-			bf.reset();
+		if (!batchfiles.empty() && !call) {
+			batchfiles.pop();
 		}
 
 		auto reader = FileReader::GetFileReader(fullname);
 		if (reader) {
-			bf = std::make_shared<BatchFile>(this,
-			                                 std::move(*reader),
-			                                 std::string(name).c_str(),
-			                                 std::string(args).c_str());
+			batchfiles.emplace(this,
+			                   std::move(*reader),
+			                   std::string(name).c_str(),
+			                   std::string(args).c_str());
```

Comment:
The `std::string(string_view).c_str()` looks pretty convoluted (Getting a C-string from a temporarily constructed `std::string` initialized with a `std::string_view`).

If the `BatchFile constructor takes in a std::string, then I think we should be able to construct with the sting_view straight-away (the std::string will take a copy of the string_view and then terminate them with `\0`, right?)